### DISCUSSION TITLE
C# script: retrieve Model current location

### DISCUSSION THREAD

**Discussion by rem-bou** [TabularEditor/TabularEditor3#1030] [Ideas]
Hi team,

Is it possible to retrieve the current location of a .bim file or serialised files?
I am interested to get the model's relative path to be able to do a number of checks and importing BPA rules, etc...

I wasn't able to find the answer by mysefl after searching on internet and trying ChatGPT (why not right?).

**Comment by mlonsk** [comment]
Hi @rem-bou 
There is not, currently, a public API in TE to do this with. 
We have the workaround specified bellow, but this means that we cannot guarantee support in future versions. 
```CSharp
var handlerProperty = Model.GetType().GetProperty("Handler", System.Reflection.BindingFlags.Instance | System.Reflection.BindingFlags.NonPublic);
var handler = handlerProperty.GetValue(Model);
var sourceProperty = handler.GetType().GetProperty("Source");
var source = sourceProperty.GetValue(handler) as string;
Info(source);
```
This script only works with models that are only connected to a .bim file or folder. I.e. it does not give you the path of the file if you are in workspace mode, rather the path of the workspace. 

We have it on our backlog to support these types of APIs in the future

**Comment by rem-bou** [comment]
Hi @mlonsk 

Thank you for the work around and putting that on your backlog!
Will have to wait for the API call to be supported as we mainly work in workspace mode. but in the rare cases we are not, that will help grealy decrease user interaction.
